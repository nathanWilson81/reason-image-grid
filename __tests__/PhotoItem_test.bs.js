// Generated by BUCKLESCRIPT, PLEASE EDIT WITH CARE
'use strict';

var Jest = require("@glennsl/bs-jest/src/jest.js");
var React = require("react");
var JestDom = require("bs-jest-dom/src/JestDom.bs.js");
var ReactTestingLibrary = require("bs-react-testing-library/src/ReactTestingLibrary.bs.js");
var PhotoItem$ReactHooksTemplate = require("../src/PhotoItem.bs.js");

Jest.describe("PhotoItem", (function (param) {
        var photo = {
          albumId: 1,
          id: 1,
          title: "This is the title",
          url: "foo1",
          thumbnailUrl: "foo2"
        };
        var photos = /* :: */[
          photo,
          /* [] */0
        ];
        var setModalOpen = (jest.fn());
        Jest.test("renders the item", (function (param) {
                return Jest.Expect.toMatchSnapshot(Jest.Expect.expect((function (eta) {
                                    return ReactTestingLibrary.render(undefined, undefined, eta);
                                  })(React.createElement(PhotoItem$ReactHooksTemplate.make, {
                                        photo: photo,
                                        photos: photos,
                                        setModalOpen: setModalOpen
                                      })).container));
              }));
        return Jest.test("sets the src on the image", (function (param) {
                      return JestDom.toHaveAttribute("src", "foo1", Jest.Expect.expect(ReactTestingLibrary.getByAltText("", (function (eta) {
                                              return ReactTestingLibrary.render(undefined, undefined, eta);
                                            })(React.createElement(PhotoItem$ReactHooksTemplate.make, {
                                                  photo: photo,
                                                  photos: photos,
                                                  setModalOpen: setModalOpen
                                                })))));
                    }));
      }));

/*  Not a pure module */
