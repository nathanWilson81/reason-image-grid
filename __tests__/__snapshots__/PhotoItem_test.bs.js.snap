// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoItem renders the item 1`] = `
<div>
  <div
    class="css-5v12vm"
  >
    <img
      alt=""
      class="css-1ye53x6"
      src="foo1"
    />
  </div>
</div>
`;
